File: uva/10226 Hardwood Species/hardwood.py
for _ in range(int(input()) + 1):
    s = input()
    counts = {}
    total = 0
    while s != "":
        total += 1
        counts[s] = (counts.get(s) or 0) + 1
        try:
            s = input()
        except:
            break
    for s in sorted(counts):
        print(s, "{:.4f}".format(counts[s] / total * 100))

File: uva/10226 Hardwood Species/hardwood.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int t;
    cin >> t;
    string trash;
    getline(cin, trash);
    getline(cin, trash);
    while (t --) {
        string s;
        map<string, int> counts;
        double total = 0;
        while (getline(cin, s) and s != "") {
            counts[s] ++;
            total ++;
        }
        for (auto tmp: counts) {
            s = tmp.first;
            cout << s << " " << fixed << setprecision(4) << counts[s] / total * 100.0 << endl;
        }
        if (t) {
            cout << endl;
        }
    }
}